test/language/number/scientific_literals.wren

```diff
@@ -0,0 +1,9 @@
+var x = 2.55e2
+
+IO.print(x) // expect: 255
+IO.print(x + 1) // expect: 256
+IO.print(x == 255) // expect: true
+IO.print(2.55e-2 is Num) // expect: true
+IO.print(x is Num) // expect: true
+IO.print(-2.55e2) // expect: -255
+IO.print(-25500e-2) // expect: -255
```

Comment:
These tests are great!

Can you add a couple of error case tests too? You'll probably need to make each one a separate tiny file. I'm thinking things like:

```
1e
1.e
1.2e
1.2e3e
1.2e3.4
```

Also, two tests like "literal_too_large.wren" to test how it handles an overly long exponent and an overly long mantissa.

I know that's a lot of corner cases, but I think it's really really important to have very detailed test coverage. It's one of the things we can do to give people confidence in embedding Wren in their programs.